### PR TITLE
Hypre build variants

### PR DESCRIPTION
**Pull request by PaulMullowney**
When building with rocm, one needs to build hypre with rocm. Adding device blas libraries as variants.

### CODE REVIEW

**Comment by psakievich** [comment]
@PaulMullowney I'm assuming you are adding these as variants because you still want the option to use non-device blas?  If not then we don't need to add them as variants. You can just trigger on `+cuda/+rocm`.

File: repos/exawind/packages/hypre/package.py [hunk 1]
```diff
@@ -7,6 +7,8 @@ class Hypre(bHypre):
     phases = ['autoreconf', 'distclean', 'configure', 'clean', 'build', 'install']
 
     variant('gpu-aware-mpi', default=False, description='Use gpu-aware mpi')
+    variant('rocblas', default=False, description='use rocblas')
+    variant('cublas', default=False, description='use cublas')
```
Comment (psakievich) [inline repos/exawind/packages/hypre/package.py:11]:
```suggestion
    variant('cublas', default=False, description='use cublas')
    conflict('+cublas', when='~cuda', msg='cublas requires cuda to be enabled')
    conflict('+rocblas', when='~rocm', msg='rocblas requires rocm to be enabled')
```
We should add some safety rails to this.

**Review by psakievich — CHANGES_REQUESTED** [review]

File: repos/exawind/packages/hypre/package.py (hunk 1 above)
Comment (PaulMullowney) [reply repos/exawind/packages/hypre/package.py:11]: I like the safety checks

**Comment by PaulMullowney** [comment]
> @PaulMullowney I'm assuming you are adding these as variants because you still want the option to use non-device blas? If not then we don't need to add them as variants. You can just trigger on `+cuda/+rocm`.

Actually, you don't get them when triggering +cuda/+rocm. Hypre does not do this by default. I want the option of using them.

File: repos/exawind/packages/hypre/package.py (hunk 1 above)
Comment (PaulMullowney) [reply repos/exawind/packages/hypre/package.py:11]: I see the following in the tests:
rhodes failed to concretize. Additional exception: name 'conflict' is not defined

Comment (psakievich) [reply repos/exawind/packages/hypre/package.py:11]: whoops it's `conflicts` with an `s` my bad.

**Review by psakievich — APPROVED** [review]

**Pull request merged**